arraycontext/impl/jax/__init__.py

```diff
@@ -0,0 +1,132 @@
+"""
+.. currentmodule:: arraycontext
+.. autoclass:: EagerJAXArrayContext
+.. autoclass:: JITCompilingJAXArrayContext
+"""
+
+__copyright__ = """
+Copyright (C) 2021 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+
+from typing import Sequence, Union, Callable, Any
+from pytools.tag import Tag
+from arraycontext.context import ArrayContext, _ScalarLike
+
+
+class EagerJAXArrayContext(ArrayContext):
+    """
+    A :class:`ArrayContext` that uses :mod:`jax.numpy.DeviceArray` instances
+    for its base array class. Performs all array operations eagerly. See
+    :class:`~arraycontext.PytatoJAXArrayContext` for a lazier version.
+
+    .. note::
+
+        JAX stores a global configuration state in the module
+        :mod:`jax.config`. Callers are expected to maintain those. Most
+        important for scientific computing workloads being
+        ``jax_enable_x64``.
```

Comment:
This doesn't seem like it will be relevant for much longer
https://github.com/google/jax/issues/8178

Reply:
Since that's still not resolved. I'll keep the note at this point. (Thanks!)

Reply:
Yeah, definitely. I hoped they'd fix it by now, but it looks like it stalled :cry: